teemaderegister-be/src/controllers/admin.js

```diff
@@ -1,5 +1,55 @@
+const User = require('../models/user')
+const slug = require('slug')
+const crypto = require('crypto')
+const mail = require('./../utils/mail')
+
+module.exports.createUser = async (req, res) => {
+  const {firstName, lastName, email, role} = req.body
+  const slugs = `${slug(firstName)}-${slug(lastName)}`
+
+  const alreadyUser = await User.findOne({'login.email': email})
+  if (alreadyUser) {
+    return res.status(406).json({message: 'Email already in use!', success: 0})
+  }
+
+  const user = {
+    profile: {
+      firstName: firstName,
+      lastName: lastName,
+      slug: slugs
+    },
+
+    login: {
+      email: email,
+      passwordResetToken: crypto.randomBytes(20).toString('hex'),
+      passwordResetExpires: Date.now() + parseInt(process.env.PASSWORD_RESET_TOKEN_EXPIRES),
```

Comment:
This logis is used also in auth controlles. Would just propose exporting it also from auth controller and reusing it here and also in auth controlles. E.g something like this
```JS
module.exports.getPasswordResetTokenValues= () => {
  return  {
    passwordResetToken: crypto.randomBytes(20).toString('hex'),
    passwordResetExpires: Date.now() + parseInt(process.env.PASSWORD_RESET_TOKEN_EXPIRES)
  }
}
```